pkg/client/core/service_cluster_client.go

```diff
@@ -15,15 +15,15 @@ import (
 )
 
 type clusterService struct {
-	api   client.ClusterAPI
-	state client.ClusterState
+	service api.ClusterService
+	state   client.ClusterState
 
 	provider v1alpha1.CloudProvider
 	auth     aws.Authenticator
 }
 
 func (s *clusterService) CreateCluster(_ context.Context, opts client.ClusterCreateOpts) (*client.Cluster, error) {
-	c, err := s.api.CreateCluster(api.ClusterCreateOpts{
+	c, err := s.service.CreateCluster(context.Background(), api.ClusterCreateOpts{
```

Comment:
Use the context argument available from the function signature

Reply:
Yepp, have done this in most cases, nice catch! Will fix